testing/validation.sh

```diff
@@ -0,0 +1,177 @@
+}
+
+function resetLogFile {
+    > $LOG_FILE
+}
+
+function validateOrkestraDeployment {
+    resources=$(kubectl get pods --namespace orkestra &>> $LOG_FILE | grep -i -c running)
+    if [ $resources -ne $ORKESTRA_RESOURCE_COUNT ]; then
+        testSuiteMessage "TEST_FAIL" "No running orkestra resources. Currently $resources running resources. Expected $ORKESTRA_RESOURCE_COUNT"
+    else
+        testSuiteMessage "TEST_PASS" "orkestra resources are running"
+    fi
+
+    orkestraStatus=$(helm status orkestra -n orkestra &>> $LOG_FILE | grep -c deployed)
+    if [ $orkestraStatus -eq 1 ]; then
+        testSuiteMessage "TEST_PASS" "orkestra deployed successfully"
+    else
+        testSuiteMessage "TEST_FAIL" "orkestra not deployed"
+    fi
+}
+
+function validateBookInfoDeployment {
+    ambassadorStatus=$(helm status ambassador -n ambassador &>> $LOG_FILE | grep -c deployed)
+    if [ $ambassadorStatus -eq 1 ]; then
+        testSuiteMessage "TEST_PASS" "ambassador deployed successfully"
+    else
+        testSuiteMessage "TEST_FAIL" "ambassador not deployed"
+    fi
+
+    bookinfoReleaseNames=("details" "productpage" "ratings" "reviews" "bookinfo")
+
+    for var in "${bookinfoReleaseNames[@]}"
+    do  
+        deployedStatus=$(helm status $var -n bookinfo &>> $LOG_FILE | grep -c deployed)
+        if [ $deployedStatus -eq 1 ]; then
+            testSuiteMessage "TEST_PASS" "$var deployed successfully"
+        else
+            testSuiteMessage "TEST_FAIL" "$var not deployed"
+        fi
+    done
+}
+
+function validateArgoWorkflow {
+    bookinfoStatus=$(curl -s --request GET --url http://localhost:2746/api/v1/workflows/orkestra/bookinfo | grep -c "not found")
+    if [ "$bookinfoStatus" -eq 1 ]; then
+        testSuiteMessage "TEST_FAIL" "No argo workflow found for bookinfo"
+    else
+        argoNodes=($(curl -s --request GET --url http://localhost:2746/api/v1/workflows/orkestra/bookinfo | jq -c '.status.nodes[] | {id: .id, name: .name, displayName: .displayName, phase: .phase}'))
+
+        requiredNodes=(
+            "bookinfo" 
+            "bookinfo.bookinfo.ratings" 
+            "bookinfo.ambassador" 
+            "bookinfo.bookinfo.details"
+            "bookinfo.bookinfo.productpage"
+            "bookinfo.ambassador.ambassador"
+            "bookinfo.bookinfo.reviews"
+            "bookinfo.bookinfo.bookinfo"
+            "bookinfo.bookinfo"
+        )
+
+        for node in "${requiredNodes[@]}"
+        do
+            status=$(curl -s --request GET --url http://localhost:2746/api/v1/workflows/orkestra/bookinfo | jq --arg node "$node" -r '.status.nodes[] | select(.name==$node) | .phase')
+            if [ "$status" == "Succeeded" ]; then
+                testSuiteMessage "TEST_PASS" "argo node: $node has succeeded"
+            else
+                testSuiteMessage "TEST_FAIL" "$node status: $status, Expected Succeeded"
+            fi
+        done
+    fi
+}
+
+function validateApplicationGroup {
```

Comment:
We should be checking the `.spec.applications[].status` and not .spec.applications`. Since checking the spec will always pass, as it is just verifying that the sample `bookinfo` has the expected applications configured in it's spec.

Unless I am misunderstanding this validation test entirely.

I do see that you do status checks and argo workflow checks in the rest of the validate functions so we should do something similar for the ApplicationGroup status and applications status